### DISCUSSION TITLE
Jetson Nano can't communicate with Create 3

### DISCUSSION THREAD

**Discussion by jungseokhong** [iRobotEducation/create3_docs#307] [Questions]
I am using Jetson Nano 4GB version to connect to create 3 via USB.
I've followed steps in https://iroboteducation.github.io/create3_docs/setup/jetson/.

Basically, using docker container to install ros 2.

In the step 4 (Set the default network interface by setting Cyclone DDS configuration), 
I tried to 
`export CYCLONEDDS_URI='<CycloneDDS><Domain><General><NetworkInterfaceAddress>l4tbr0</NetworkInterfaceAddress></General></Domain></CycloneDDS>'
`
in docker container. (also tried to it in the jetson nano host. but it didn't work either)
ros2 topic list doesn't show me any create 3 topics lists. 
Is there any suggestions that I can try?

I don't know how to check the connections between create 3 and jetson nano as well.

**Comment by alsora** [comment]
The environment variable `CYCLONEDDS_URI` must be exported in the same shell session where you are then running `ros2 topic list`. This means that you must run that command inside the Docker container.

Can you try to use a more robust command to communicate with the robot:
```
ros2 topic list --no-daemon --spin-time 10
```

An alternative option could be to use FastDDS rather than CycloneDDS. This should offer an easier experience out of the box.
You need to make this change on the robot through the webserver (https://iroboteducation.github.io/create3_docs/webserver/application/#application-configuration) and also inside the Docker container by running
```
export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
ros2 topic list --no-daemon --spin-time 10
```

Let me know if this helps!

**Reply by jungseokhong** [reply]
Thank you for the answer. When I used rmw_fastrtps_cpp with USB C (create 3) and Ethernet (nano), it worked. My understanding is that I don't have to set anything specific (`eth0`, `wlan0`, etc) when I used rmw_fastrtps_cpp, correct? so if I use another machine connected under same wifi, I am guessing I should be able to access both create 3 and nano.

 Is there any reason that CYCLONEDDS_URI was recommended in the tutorial?
I haven't tested your commands with USB C (create 3) and micro USB (nano) option yet. do you recommend USB c - micro USB or USB c - ethernet?

**Reply by alsora** [reply]
Yes, Fast-DDS by default discovers all available network interfaces, differently from cyclonedds where interfaces must be manually specified.

The documentation mentions cyclonedds because that's the default for a ros galactic installation.
The procedure described there should work, but it's more complex and easier to get it wrong that just using fastdds.

I'm not an expert of connecting compute boards, but I use USB C  on both sides in my setup.
It's important to make sure that the cable supports data (and not only power)

**Reply by jungseokhong** [reply]
My jetson nano 4GB model does not have a USB C port to communicate. That's why I had to use micro USB or Ethernet. Even when you use USB C - USB C connection. did you have to set the static IP for the jetson (l4tbr0)?
I am guessing it won't work without it.

**Reply by jungseokhong** [reply]
Just tried with micro USB from Nano to create 3 USB C. but it didn't work for me. although I was able to ping to 192.168.186.2.

Then i tried again with my original ethernet setup after disabling the static IP address for micro usb by this : 
`sudo mv /opt/nvidia/l4t-usb-device-mode/IP_ADDRESS_FOR_CREATE3_ROBOT.conf /opt/nvidia/l4t-usb-device-mode/IP_ADDRESS_FOR_CREATE3_ROBOT.conf.bak`
from this tutorial (https://iroboteducation.github.io/create3_docs/setup/jetson/).

I was able to ping 192.168.186.2. However, even with ethernet it didn't work.

for both of above cases, I used
```
export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
ros2 topic list --no-daemon --spin-time 10
```

I don't know what I am missing but not sure why ethernet method is not working anymore..
Is there anything that I can try?
I double checked the robots setting from 196.168.10.1
